src/onboarding.nim

```diff
@@ -0,0 +1,147 @@
+import NimQml
+import json
+import status/accounts
+import nimcrypto
+import status/utils
+import status/libstatus
+import models/accounts as Models
+import constants/constants
+import uuids
+import eventemitter
+import status/test as status_test
+
+# Probably all QT classes will look like this:
+QtObject:
+  type Onboarding* = ref object of QObject
+    m_generatedAddresses: string
+    events: EventEmitter
+
+  # ¯\_(ツ)_/¯ dunno what is this
+  proc setup(self: Onboarding) =
+    self.QObject.setup
+
+  # ¯\_(ツ)_/¯ seems to be a method for garbage collection
+  proc delete*(self: Onboarding) =
+    self.QObject.delete
+
+  # Constructor
+  proc newOnboarding*(events: EventEmitter): Onboarding =
+    new(result, delete)
+    result.events = events
+    result.setup()
+
+  # Read more about slots and signals here: https://doc.qt.io/qt-5/signalsandslots.html
+
+  # Accesors
+  proc getGeneratedAddresses*(self: Onboarding): string {.slot.} =
+    result = self.m_generatedAddresses
+
+  proc generatedAddressesChanged*(self: Onboarding,
+      generatedAddresses: string) {.signal.}
+
+  proc setGeneratedAddresses*(self: Onboarding, generatedAddresses: string) {.slot.} =
+    if self.m_generatedAddresses == generatedAddresses:
+      return
+    self.m_generatedAddresses = generatedAddresses
+    self.generatedAddressesChanged(generatedAddresses)
+
+  QtProperty[string]generatedAddresses:
+    read = getGeneratedAddresses
+    write = setGeneratedAddresses
+    notify = generatedAddressesChanged
+
+  # QML functions
+  proc generateAddresses*(self: Onboarding) {.slot.} =
+    self.setGeneratedAddresses(generateAddresses())
+
+  proc generateAlias*(self: Onboarding, publicKey: string): string {.slot.} =
+    result = $libstatus.generateAlias(publicKey.toGoString)
```

Comment:
We should not call libstatus directly from these QObjects, but create procs inside the folder `status/` that expose these functions. This is so it's easier to move from status-go to nimbus in the future. (We'd only have to change the procs inside status/ instead of hunting these calls in the rest of the code)